### ISSUE TITLE
dependency: provide access to dependency inputs

### ISSUE THREAD

**Issue by lsq-ablair** [gruntwork-io/terragrunt#2388]
When a dependency is specified, we can use the `outputs` of that dependency in the current module.

However, it would be "nice" to be able to use the `inputs` of that module as well. If we consider that we specify, perhaps, a date or time, or certain Tag values to the module, those values might not be represented in the module's outputs. However, a dependency may want to know those values -- so that something can be created with the same time interval, or with the same tags as the parent.

```
terraform {
  source = "email-notifications"
}
dependency "cluster" {
  config_path = "../cluster"
}

inputs {
  cluster_name = dependency.cluster.outputs.name

  addresses = [ for u in dependency.cluster.inputs.users : "${u}@mydomain.com" ]

  aws_tags = merge(dependency.cluster.inputs.aws_tags, { "module" = "email-notifications" })
}
```
in this case, "users" would be an input to "cluster" (probably authentication/permissions) and I would to set them up for e-mail notifications about the cluster in a separate module.

**Comment by bkalcho** [comment]
+1

**Comment by levkohimins** [comment]
Resolved [v0.55.4](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.55.4) release.
